notus/scanner/models/packages/package.py

```diff
@@ -98,6 +102,69 @@ def __hash__(self) -> int:
         # the full name identifies the package
         return hash(self.full_name)
 
+    @staticmethod
+    def version_compare(version_a: str, version_b: str) -> PackageComparison:
+        """Compares two versions"""
+        if version_a == version_b:
+            return PackageComparison.EQUAL
+
+        a_parts = version_component_re.split(version_a)
+        b_parts = version_component_re.split(version_b)
+        pos = 0
+
+        while True:
+            try:
+                a_part = a_parts[pos]
```

Comment:
I think a len comparison would be better than to relay on IndexError while trying to access it